### DISCUSSION TITLE
Can Cowboy used as cdn Edge cache?

### DISCUSSION THREAD

**Discussion by johnynixons** [ninenines/cowboy#1644] [General]
I am looking for high performance CDN software.  since i am erlang fan, i was exploring the http, https,tls implementation in erlang.
For further details ping me.  we can workout on this.

**Comment by essen** [comment]
In the sense that a CDN edge cache is in part an HTTP server, yes. And Gun could be the client part doing the non-cached requests. But a lot of work is needed on top of that, and most of that work is outside of the scope of Cowboy (and should be done outside Cowboy).

**Reply by johnynixons** [reply]
Thank you , you mean, caching part , upstream load balancing etc.. ? If i am to compare Golang-net/http or fast/http, or Proxygen, what are the things need to be added.

**Reply by essen** [reply]
Not familiar with those. Cowboy is an HTTP server library that doesn't do much on top of implementing protocols and defining an interface to use them. So with regard to caching or proxying it'll at best have functions to parse or build related headers. But it won't do any caching or proxying, this is something you would have to implement.
